lnwallet/witnessgen.go

```diff
@@ -15,16 +15,16 @@ type WitnessType uint16
 const (
 	// CommitmentTimeLock is a witness that allows us to spend the output of a
 	// commitment transaction after a relative lock-time lockout.
-	CommitmentTimeLock WitnessType = 0
+	CommitmentTimeLock WitnessType = iota
```

Comment:
Agree that the latter sounds better as it decouples the two. Maybe with an error returned from the database code if it encounters an unknown WitnessType.